src/main/java/hudson/plugins/spotinst/cloud/BaseSpotinstCloud.java

```diff
@@ -643,6 +732,179 @@ protected Integer getPendingThreshold() {
+    //                        }
+    //                    }
+    //                }
+    //                else {
+    //                    LOGGER.warn("redis response value return null");
+    //                }
+    //            }
+    //            //there is no controller for the given group in redis, should take ownership
+    //            else {
+    //                handleGroupManagedByThisController(cloud, controllerIdentifier);
+    //            }
+    //        }
+    //    }
+
+    public void syncGroupsOwner(BaseSpotinstCloud cloud) {
+        String groupId   = cloud.getGroupId();
+        String accountId = cloud.getAccountId();
+        LOGGER.info(String.format("try fetching controller identifier for group %s from redis", groupId));
+
+        IRedisRepo          redisRepo             = RepoManager.getInstance().getRedisRepo();
+        ApiResponse<Object> redisGetValueResponse = redisRepo.getValue(groupId, accountId);
+        String              controllerIdentifier  = SpotinstContext.getInstance().getControllerIdentifier();
+
+        if (redisGetValueResponse.isRequestSucceed()) {
+            //redis response might return in different types
+            if (redisGetValueResponse.getValue() instanceof String) {
+                String redisResponseValue = (String) redisGetValueResponse.getValue();
+
+                if (redisResponseValue != null) {
+                    boolean isGroupBelongToController = redisResponseValue.equals(controllerIdentifier);
+
+                    if (isGroupBelongToController) {
+                        handleGroupManagedByThisController(cloud, controllerIdentifier);
+                    }
+                    else {
+                        LOGGER.info(String.format("group %s does not belong to controller with identifier %s", groupId,
+                                                  controllerIdentifier));
+                        GroupStateTracker groupStateDetails =
+                                SpotinstContext.getInstance().getConnectionStateByGroupId().get(groupId);
+
+                        if (groupStateDetails == null) {
+                            groupStateDetails = new GroupStateTracker();
+                            groupStateDetails.setGroupId(groupId);
+                            groupStateDetails.setAccountId(accountId);
+                            groupStateDetails.setState(SPOTINST_CLOUD_COMMUNICATION_INITIALIZING);
+                            Date now = new Date();
+                            groupStateDetails.setTimeStamp(now);
+                            SpotinstContext.getInstance().getConnectionStateByGroupId().put(groupId, groupStateDetails);
+                        }
+                    }
+                }
+            }
+            else {
+                LOGGER.warn("redis response value return null");
+            }
+        }
+        //there is no controller for the given group in redis, should take ownership
+        else {
```

Comment:
This should be for failure during reading from redis only, just write the error and the current status to log and stay in the current status

Reply:
done